compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2592,8 +2592,51 @@ public void visit(BLangXMLQName bLangXMLQName) {
             bLangXMLQName.type = symTable.semanticError;
             return;
         }
-        bLangXMLQName.namespaceURI = ((BXMLNSSymbol) xmlnsSymbol).namespaceURI;
-        bLangXMLQName.nsSymbol = (BXMLNSSymbol) xmlnsSymbol;
+
+        if (xmlnsSymbol.getKind() == SymbolKind.PACKAGE) {
+            xmlnsSymbol = getXmlNsSymbolFromPkgSymbol(bLangXMLQName.localname.value, bLangXMLQName.prefix.value,
+                    xmlnsSymbol, bLangXMLQName.pos);
+        }
+        if (xmlnsSymbol != null && xmlnsSymbol.getKind() == SymbolKind.XMLNS) {
+            bLangXMLQName.namespaceURI = ((BXMLNSSymbol) xmlnsSymbol).namespaceURI;
+            bLangXMLQName.nsSymbol = (BXMLNSSymbol) xmlnsSymbol;
+        } else {
+            resultType = symTable.semanticError;
+            return;
+        }
+    }
+
+    private BSymbol getXmlNsSymbolFromPkgSymbol(String localname, String prefix, BSymbol xmlnsSymbol, DiagnosticPos pos) {
+        BPackageSymbol pkgSymbol = (BPackageSymbol) xmlnsSymbol;
+        BSymbol constSymbol = symResolver.lookupMemberSymbol(pos, pkgSymbol.scope, env,
+                names.fromString(localname), SymTag.CONSTANT);
+        if (constSymbol != symTable.notFoundSymbol) {
+            BConstantSymbol constantSymbol = (BConstantSymbol) constSymbol;
+            if (constantSymbol.literalType.tag != TypeTags.STRING) {
+                dlog.error(pos, DiagnosticCode.INCOMPATIBLE_TYPES,
+                        symTable.stringType, constantSymbol.literalType);
+                resultType = symTable.semanticError;
+                return null;
+            } else {
+                String constVal = (String) constantSymbol.value.value;
+                int s = constVal.indexOf('{');
+                int e = constVal.lastIndexOf('}');
+                if (e > s) {
```

Comment:
I'm assuming this logic is for constants of the form `"{http://www.w3.org/XML/1998/namespace}space"`? If so, is it a possibility to have multiple braces within such a string? In that case, would the result produced by this logic still be correct?

Reply:
Yey, it's URI, not an HTTP URL